src/rocdecode/roc_decoder_caps.h

```diff
@@ -29,80 +29,214 @@ THE SOFTWARE.
-    std::vector<int> bitdepth_minus8;
+    int max_bit_depth;
     uint16_t output_format_mask;
     uint32_t max_width;
     uint32_t max_height;
     uint16_t min_width;
     uint16_t min_height;
 };
 
-// The VcnCodecsSpec struct contains information for all supported codecs and number of vcn instances per device
-struct VcnCodecsSpec {
-    std::unordered_map<rocDecVideoCodec, CodecSpec> codecs_spec;
-    uint8_t num_decoders;
-};
-
 // The RocDecVcnCodecSpec singleton class for providing access to the the vcn_spec_table
 class RocDecVcnCodecSpec {
 public:
     static RocDecVcnCodecSpec& GetInstance() {
         static RocDecVcnCodecSpec instance;
         return instance;
     }
-    rocDecStatus GetDecoderCaps(std::string gcn_arch_name, RocdecDecodeCaps *pdc) {
-        std::lock_guard<std::mutex> lock(mutex);
-        std::size_t pos = gcn_arch_name.find_first_of(":");
-        std::string gcn_arch_name_base = (pos != std::string::npos) ? gcn_arch_name.substr(0, pos) : gcn_arch_name;
-        auto it = vcn_spec_table.find(gcn_arch_name_base);
-        if (it != vcn_spec_table.end()) {
-            const VcnCodecsSpec& vcn_spec = it->second;
-            auto it1 = vcn_spec.codecs_spec.find(pdc->codec_type);
-            if (it1 != vcn_spec.codecs_spec.end()) {
-                const CodecSpec& codec_spec = it1->second;
-                auto it_chroma_format = std::find(codec_spec.chroma_format.begin(), codec_spec.chroma_format.end(), pdc->chroma_format);
-                auto it_bitdepth_minus8 = std::find(codec_spec.bitdepth_minus8.begin(), codec_spec.bitdepth_minus8.end(), pdc->bit_depth_minus_8);
-                if (it_chroma_format != codec_spec.chroma_format.end() && it_bitdepth_minus8 != codec_spec.bitdepth_minus8.end()) {
-                    pdc->is_supported = 1;
-                    pdc->num_decoders = vcn_spec.num_decoders;
-                    pdc->output_format_mask = codec_spec.output_format_mask;
-                    pdc->max_width = codec_spec.max_width;
-                    pdc->max_height = codec_spec.max_height;
-                    pdc->min_width = codec_spec.min_width;
-                    pdc->min_height = codec_spec.min_height;
-                    return ROCDEC_SUCCESS;
-                } else {
-                    return ROCDEC_NOT_SUPPORTED;
+    rocDecStatus ProbeHwDecodeCapabilities() {
+        std::string drm_node = "/dev/dri/renderD128"; // look at device_id 0
+        int drm_fd = open(drm_node.c_str(), O_RDWR);
+        if (drm_fd < 0) {
+            ERR("Failed to open drm node." + drm_node);
+            return ROCDEC_DEVICE_INVALID;
+        }
+        VADisplay va_display = vaGetDisplayDRM(drm_fd);
+        if (!va_display) {
+            ERR("Failed to create va_display.");
+            return ROCDEC_DEVICE_INVALID;
+        }
+        int major_version = 0, minor_version = 0;
+        CHECK_VAAPI(vaInitialize(va_display, &major_version, &minor_version));
```

Comment:
call `vaSetInfoCallback(va_display_, NULL, NULL);` before calling the `vaInitialize` to silence the VA-API for printing out the below messages when running any test.

```
libva info: VA-API version 1.16.0
libva info: Trying to open /opt/amdgpu/lib/x86_64-linux-gnu/dri/radeonsi_drv_video.so
libva info: Found init function __vaDriverInit_1_16
libva info: va_openDriver() returns 0
```

Reply:
I don't see the above messages. But will add to make sure.